group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -4007,41 +3968,21 @@ public void testDynamicMembersJoinGroupWithMaxSizeAndRequiredKnownMember() {
 
         // First round of join requests. Generate member ids. All requests will be accepted
         // as the group is still Empty.
-        List<CompletableFuture<JoinGroupResponseData>> firstRoundFutures = new ArrayList<>();
-        IntStream.range(0, groupMaxSize + 1).forEach(i -> {
-            CompletableFuture<JoinGroupResponseData> responseFuture = new CompletableFuture<>();
-            firstRoundFutures.add(responseFuture);
-            CoordinatorResult<Void, Record> result = context.sendGenericGroupJoin(request, responseFuture, requiredKnownMemberId);
-            assertTrue(responseFuture.isDone());
-            JoinGroupResponseData response = null;
-            try {
-                response = responseFuture.get();
-            } catch (Exception ignored) {
-            }
-            assertNotNull(response);
-            assertEquals(Errors.MEMBER_ID_REQUIRED.code(), response.errorCode());
-            assertTrue(result.records().isEmpty());
-        });
+        List<JoinResult> firstRoundJoinResults = IntStream.range(0, groupMaxSize + 1)
+            .mapToObj(i -> context.sendGenericGroupJoin(request, requiredKnownMemberId)).collect(Collectors.toList());
```

Comment:
nit: Should we format it as follow to keep the style consistent with the following Stream API usaged?

```
List<JoinResult> firstRoundJoinResults = IntStream.range(0, groupMaxSize + 1).mapToObj(i -> context.sendGenericGroupJoin(
    request,
    requiredKnownMemberId
)).collect(Collectors.toList());
```

alternatively, we could just put `.collect(...)` on a new line. I am fine either ways.